lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -324,6 +324,9 @@ public JsonObject asJSON() {
 	 * @param callback (err, tokenDetails)
 	 */
 	public TokenDetails authorise(AuthOptions options, TokenParams params) throws AblyException {
+		/* Spec: RSA10g */
+		storeAuthOptions(options);
```

Comment:
I'd prefer if we have something like:

.... in `AuthOptions` :

```
AuthOptions storedValues() {
  AuthOptions result = new AuthOptions();
  result.key = this.key;
  result.authUrl = authUrl;
  ...
}
```

and then here:

```
  this.authOptions = options.storedValues();
  this.tokenParams = params.storedValues();
```

Then at least we don't need to leak too much dependency between those types and this class.